app/api/orders.py

```diff
@@ -535,3 +533,30 @@ def omise_checkout(order_identifier):
         logging.info(f"Successful charge: {charge.id}.  Order ID: {order_identifier}")
 
         return redirect(make_frontend_url('orders/{}/view'.format(order_identifier)))
+
+
+@order_misc_routes.route('/orders/<string:order_identifier>/resend-email', methods=['POST'])
+def resend_emails(order_identifier):
+    """
+    Sends confirmation email for pending and completed orders on organizer request
+    :param order_identifier:
+    :return: JSON response if the email was succesfully sent
+    """
+    order = safe_query(db, Order, 'identifier', order_identifier, 'identifier')
```

Comment:
Sometimes, the order may not be fetched. So I guess a try...except block would be great here :D

Reply:
The failure is simply handled by a server 500 in the `safe_query` itself, IMO that should do